docs/developer-docs/smart-contracts/overview/trust-in-canisters.mdx

```diff
@@ -36,26 +34,25 @@ For critical applications like those used in DeFi, centralized mutability can be
 Canister controllers, if not voluntarily decentralized, have complete control over the user assets held by the canister, such as any ICP tokens or bitcoin held by the canister on the user's behalf. The controller, if malicious, can steal these assets. In other words, as a user, if you interact with a canister that deals with your assets, inspect the canister to know how it handles them. If you determine that the canister is storing assets in its subaccounts, ensure that the canister controller is decentralized.
 :::
 
-## Sole control by the Network Nervous System (NNS)
-
-The simplest option is to remove a canister's controller. Without a controller, the canister can only be mutated by the NNS via NNS proposal, assuming the integrity of the platform is maintained.
+## Verifying a canister's controllers
 
 A user can verify the list of controllers for a canister using [`dfx canister info`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-info):
 
 ```
-dfx canister info ryjl3-tyaaa-aaaaa-aaaba-cai --network ic
+dfx canister info <canister-id> --network ic
 ```
 
-This will return the list of controllers for the canister with principal `ryjl3-tyaaa-aaaaa-aaaba-cai` (in this example, the ledger canister).
+A user can also obtain the list of a canister's controllers via a [`read_state` request](/references/ic-interface-spec.md/#http-read-state) to get the relevant [canister information](/references/ic-interface-spec.md#state-tree-canister-information), which includes the list of controllers. A canister cannot obtain this information.
+
+If a canister does not have any controllers, it can only be uninstalled by the NNS via NNS proposal, assuming the integrity of the platform is maintained.
```

Comment:
```suggestion
If a canister does not have any controllers, it cannot be changed and is thus immutable. It can only be uninstalled by the NNS via NNS proposal, assuming the integrity of the platform is maintained.
```